src/wallet/wallet.cpp

```diff
@@ -4093,3 +4095,82 @@ CKeyID CWallet::GetKeyForDestination(const CTxDestination& dest) const
     }
     return CKeyID();
 }
+
+static const std::string STYLE_STRING_NONE = "";
+static const std::string STYLE_STRING_LEGACY = "legacy";
+static const std::string STYLE_STRING_P2SH = "p2sh";
+static const std::string STYLE_STRING_SEGWIT = "segwit";
+
+OutputStyle ParseStyle(const std::string& style)
+{
+    if (style == STYLE_STRING_LEGACY) {
+        return STYLE_LEGACY;
+    } else if (style == STYLE_STRING_P2SH || style == "default") {
+        return STYLE_P2SH;
+    } else if (style == STYLE_STRING_SEGWIT) {
+        return STYLE_SEGWIT;
+    } else {
+        return STYLE_NONE;
+    }
+}
+
+
+const std::string& FormatStyle(OutputStyle style)
+{
+    switch (style) {
+    case STYLE_LEGACY: return STYLE_STRING_LEGACY;
+    case STYLE_P2SH: return STYLE_STRING_P2SH;
+    case STYLE_SEGWIT: return STYLE_STRING_SEGWIT;
+    default: assert(false);
+    }
+}
+
+std::vector<CTxDestination> CWallet::GetAllDestinationsForKey(const CPubKey& key)
+{
+    CKeyID keyid = key.GetID();
+    if (key.IsCompressed()) {
+        CTxDestination segwit = WitnessV0KeyHash(keyid);
+        CTxDestination p2sh = CScriptID(GetScriptForDestination(segwit));
+        return std::vector<CTxDestination>{std::move(keyid), std::move(p2sh), std::move(segwit)};
+    } else {
+        return std::vector<CTxDestination>{std::move(keyid)};
+    }
+}
+
+CTxDestination CWallet::GetDestinationForKey(const CPubKey& key, OutputStyle style)
```

Comment:
If IsSolvable moves out of CWallet, I believe these can too.

Reply:
Done. IsSolvable is moved to script/sign, GetDestinationForKey is moved to keystore.